### PR TITLE
Develop new facture rec field societe rib management

### PR DESCRIPTION
**Pull request by atm-lena**
# NEW|New 

Add societe rib for invoice rec. 
We can select a societe rib when we create an invoice rec if payment mode selected is "prelevement"

image

We can edit societe rib in facture rec card.

image

When an invoice has an invoice rec source with a societe rib, societe rib is preselected in prelevement tab.

image

I also modified select iban management in prelevement.php and formIban/selectTypesIban functions  to better adhere to Dolibarr standards

### CODE REVIEW

File: htdocs/compta/facture/card-rec.php [hunk 1]
```diff
@@ -356,6 +362,11 @@
 		// Set bank account
 		$object->context['actionmsg'] = $langs->trans("FieldXModified", $langs->transnoentitiesnoconv("Bank"));
 		$result = $object->setBankAccount(GETPOSTINT('fk_account'));
+	} elseif ($action == 'setbankaccountcustomer' && $usercancreate) {
+		// Set bank account customer
+		$object->context['actionmsg'] = $langs->trans("FieldXModified", $langs->transnoentitiesnoconv("BankCustomer"));
```
Comment (frederic34) [inline htdocs/compta/facture/card-rec.php:367]: you need to add missing translation `BankCustomer` in `langs/en_US/banks.lang` only

Comment (atm-lena) [reply htdocs/compta/facture/card-rec.php]: It's corrected, thank you !

File: htdocs/compta/facture/prelevement.php [hunk 2]
```diff
@@ -803,24 +799,26 @@
...
-				$ribForSelection = [];
-				$defaultRib = '';
-				foreach ($ribList as $rib) {
-					$ribString = $rib->iban . (($rib->iban && $rib->bic) ? ' / ' : '') . $rib->bic;
-					$ribForSelection[$rib->id] = $ribString;
-					if ($rib->default_rib == 1) {
-						$defaultRib = $ribString;
+				// if societe rib in model invoice, we preselect it
+				$selectedRib = '';
+				if ($object->element = 'invoice' && $object->fk_fac_rec_source) {
```
Comment (eldy) [inline htdocs/compta/facture/prelevement.php:804]: You surely want to do 
`$object->element == 'invoice'`
instead of
`$object->element = 'invoice'`

Comment (atm-lena) [reply htdocs/compta/facture/prelevement.php]: Oups ! It's done

**Pull request merged**